feat: infer missing metadata

balajtimate opened the pull request:
- Implement HTSinfer plugin in `zarp.plugins.sample_processors.htsinfer`
- Add unit tests for HTSinfer workflow in `tests/plugins/sample_processors/test_htsinfer.py`
- Update `zarp/config/models.py` with HTSinfer config file model
- Update `zarp/snakemake/run.py`, forcing singularity in case of HTSinfer workflow run

Fixes #34 

### Type of change

- [X] New feature (non-breaking change which adds functionality)

### Checklist

- [X] I have performed a self-review of my own code
- [X] My code follows the existing coding style, lints and generates no new
      warnings
- [X] I have added type annotations to all function/method signatures, and I
      have added type annotations for any local variables that are non-trivial,
      potentially ambiguous or might otherwise benefit from explicit typing.
- [X] I have commented my code in hard-to-understand areas
- [X] I have added [Google-style
      docstrings](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html)
      to all new modules, classes,
      methods/functions or updated previously existing ones
- [X] I have added tests that prove my fix is effective or that my feature
      works
- [X] New and existing unit tests pass locally with my changes and I have not
      reduced the code coverage relative to the previous state
- [X] I have updated any sections of the app's documentation that are affected
      by the proposed changes

uniqueg reviewed (changes requested):
Please remove `--htsinfer-config` arg

balajtimate commented:
Removed `--htsinfer-config` from `args.py`, `models.py` and the config yaml file.
Also added `DRY_RUN` execution mode compatibility to the HTSinfer plugin.

uniqueg commented on zarp/plugins/sample_processors/htsinfer.py:
```diff
@@ -57,7 +57,10 @@ def process(  # pylint: disable=arguments-differ
         )
         cmd = executor.compile_command(snakefile=workflow)
-        executor.run(cmd=cmd)
+        if self.config.run.execution_mode == "DRY_RUN":
+            self.records = self.records
```
This now needs it's own unit test

balajtimate replied:
Thanks for the heads-up, I added the unit test for the dry run

uniqueg reviewed (changes requested):
Looks good, but please add a unit test for the dry run
